smartclide-monitoring/src/main/java/eu/smartclide/contexthandling/services/GitRestCallService.java

```diff
@@ -0,0 +1,183 @@
+    private final String baseUri = "https://gitlab.atb-bremen.de/api/v4/projects/";
+    private final String membershipParam = "&membership=true";
+    private final String paginationParam = "&per_page=100";
+    private final String refNameParam = "&ref_name=";
+    private final String sinceParam = "&since=2022-04-25T13:05:00"; // TODO change this based on requirement
+    private final String uriParams;
+    private final String uriPartForBranches = "/repository/branches/";
+    private final String uriPartForCommits = "/repository/commits/";
+    private final String uriPartForDiff = "/diff/";
+
+    public GitRestCallService(String accessTokenParam) {
+        this.accessTokenParam = "private_token=" + accessTokenParam;
+        this.uriParams = "?" + this.accessTokenParam + this.membershipParam + this.paginationParam;
+    }
+
+    /**
+     * generates gitMessages for given user
+     * creates separate message for the commit in all the branches
+     *
+     * @return List<GitMessage>
+     */
+    public List<GitMessage> getGitMessages() {
+        // first get all user projects
+        JsonArray projects = getUserProjects();
+        List<GitMessage> gitMessages = new LinkedList<>();
+        for (JsonElement project : projects) {
+            // get project id
+            String projectId = project.getAsJsonObject().get("id").getAsString();
+            // get all branches for given project, create a new GitMessage
+            JsonArray branches = getAllBranchesForGivenProject(projectId);
+            for (JsonElement branch : branches) {
+                String branchName = branch.getAsJsonObject().get("name").getAsString();
+                // get all commits for given branch and since
+                JsonArray commitsInBranch = getCommitsForGivenBranchAndSince(projectId, branchName, sinceParam);
+                for (JsonElement commit : commitsInBranch) {
+                    String commitId = commit.getAsJsonObject().get("id").getAsString();
+
+                    GitMessage gitMessage = new GitMessage();
+                    gitMessage.setHeader(GitMessageHeader.NEW_COMMIT.getHeader());
+                    gitMessage.setState("info");
+                    gitMessage.setUser(commit.getAsJsonObject().get("author_name").getAsString());
+                    gitMessage.setRepository(project.getAsJsonObject().get("path_with_namespace").getAsString());
+                    gitMessage.setBranch(branchName);
+                    gitMessage.setTimeSinceLastCommit(calculateTimeSinceLastCommit(projectId, commit.getAsJsonObject()));
+
+                    JsonArray newCommitDiff = getCommitDiff(projectId, commitId);
+                    gitMessage.setNoOfModifiedFiles(newCommitDiff.size());
+
+                    gitMessages.add(gitMessage);
+                }
+            }
+        }
+        return gitMessages;
+    }
+
+    public Integer calculateTimeSinceLastCommit(String projectId, JsonObject newCommit) {
```

Comment:
all methods except `getGitMessages` can be `private` because they are only used inside this class.

Reply:
I had to make them `public` in order to create unit test assertions (methods which eventually creates a gitMessage).